src/core/model.js

```diff
@@ -388,7 +391,7 @@ export default class Model extends Object3D {
     log.group(LOG_DRAW_PRIORITY,
       `>>> RENDERING MODEL ${this.id}`, {collapsed: log.priority <= 2});
 
-    this.setProgramState();
+    this.setProgramState({vertexArray});
```

Comment:
Super nit: We deal with attributes above, but pass in vertexArray here. Slightly asymmetrical.

Reply:
When attributes are provided they are set on default `VertexArray` which is owned by `Program`, but if user is using custom `VertexArray`, program needs to know information about bound attribute locations.